.github/workflows/ci.yaml

```diff
@@ -0,0 +1,74 @@
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - "*"
+
+permissions:
+  deployments: read
+  contents: read
+  pull-requests: write
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          components: rustfmt
+          toolchain: nightly-2024-01-04
+      - uses: Swatinem/rust-cache@v2
+      - run: scripts/rust_fmt.sh --check
+
+  clippy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dtolnay/rust-toolchain@master
+        with:
+          components: clippy
+          toolchain: nightly-2024-01-04
+      - uses: Swatinem/rust-cache@v2
+      - run: scripts/clippy.sh
+
+  machete:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: bnjbvr/cargo-machete@main
```

Comment:
isn't this done by clippy (at least in nightly) ?

Reply:
It doesn't seem to be integrated by clippy nor rustc (see this [comment](https://www.reddit.com/r/rust/comments/ucysu7/comment/i6ecgnx))

I've quickly tried adding an unused dependency to a crate then run clippy and machete with `cargo +nightly-2024-01-04`:  machete catches the unused dep while clippy doesn't

Reply:
can we add machete to trunk then?

Reply:
There is no Trunk plugin for cargo-machete, it requires a custom linter/plugin.
Not a priority for now, I've opened an issue to tackle later on #7